elephant_vending_machine/views.py

```diff
@@ -216,12 +216,35 @@ def list_logs():
         HTTP response 200 with body {'files': [array of links to files]}.
 
     """
-    file_request_path = request.base_url[:request.base_url.rfind('/')] + "/logs/"
+    resource_route = "/logs/"
+    file_request_path = request.base_url[:request.base_url.rfind('/')] + resource_route
     path_to_current_file = os.path.dirname(os.path.abspath(__file__))
     logs_path = os.path.join(path_to_current_file, '..', 'logs')
     directory_list = os.listdir(logs_path)
-    experiment_files = [f for f in directory_list if os.path.isfile(os.path.join(logs_path, f))]
-    experiment_files.remove('.gitignore')
-    full_experiment_paths = [file_request_path + f for f in experiment_files]
+    log_files = [f for f in directory_list if os.path.isfile(os.path.join(logs_path, f))]
+    log_files.sort()
+    if '.gitignore' in log_files:
+        log_files.remove('.gitignore')
+    full_log_paths = [file_request_path + f for f in log_files]
+    response_code = 200
+    return make_response(jsonify({'files': full_log_paths}), response_code)
+
+@APP.route('/experiment', methods=['GET'])
+def list_experiments():
+    """Returns a list of experiments from the experiments directory
+
+    Returns:
+        HTTP response 200 with body {'files': [array of links to files]}.
+    """
+    resource_route = "/experiment/"
+    file_request_path = request.base_url[:request.base_url.rfind('/')] + resource_route
+    path_to_current_file = os.path.dirname(os.path.abspath(__file__))
+    experiments_path = os.path.join(path_to_current_file, 'static', 'experiments')
+    directory_list = os.listdir(experiments_path)
+    exper_files = [f for f in directory_list if os.path.isfile(os.path.join(experiments_path, f))]
+    exper_files.sort()
+    if '.gitignore' in exper_files:
+        exper_files.remove('.gitignore')
+    full_experiment_paths = [file_request_path + f for f in exper_files]
```

Comment:
Does the file_request_path include a port number? I would guess not based on your test cases below. I think our URL will end up containing the port number the webserver is running on. Can we get that value programmatically? If not, we could add a config variable for it or something.

Reply:
If the port number was included in the initial request it will be in the response (i.e. a request made to http://127.0.0.1:5000/experiment will return responses like http://127.0.0.1:5000/experiment/blank.py)